packages/mui-material/src/StepIcon/StepIcon.d.ts

```diff
@@ -1,11 +1,12 @@
 import * as React from 'react';
 import { SxProps } from '@mui/system';
-import { InternalStandardProps as StandardProps } from '..';
+import { InternalStandardProps as StandardProps, SvgIconProps } from '..';
 import { Theme } from '../styles';
 import { StepIconClasses } from './stepIconClasses';
 
 export interface StepIconProps
-  extends StandardProps<React.HTMLAttributes<HTMLDivElement>, 'children'> {
+  extends Omit<StandardProps<React.HTMLAttributes<SVGSVGElement>, 'color' | 'children'>, 'ref'>,
```

Comment:
type of ref in `StandardProps<React.HTMLAttributes<SVGSVGElement>, 'color' | 'children'>` is `React.Ref<unknown> | undefined` 

 type of `ref` in `SvgIconProps` is `((instance: SVGSVGElement | null) => void | React.DO_NOT_USE_OR_YOU_WILL_BE_FIRED_CALLBACK_REF_RETURN_VALUES[keyof React.DO_NOT_USE_OR_YOU_WILL_BE_FIRED_CALLBACK_REF_RETURN_VALUES]) | React.RefObject<...> | null | undefined`

Due to above difference and we are considering `SvgIconProps` as source as truth (since ref will eventually gets passed to `SvgIcon`), i had omitted `ref` type from `React.HTMLAttributes<SVGSVGElement>`